csharp_dotnetcore/4.Simple-Prompt-Bot/SimplePromptBot.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Builder.Dialogs;
+using Microsoft.Bot.Schema;
+
+namespace Simple_Prompt_Bot
+{
+    /// <summary>
+    /// This bot illustrates a simple prompt. In this case we are just going to ask for a name.
+    /// </summary>
+    public class SimplePromptBot : IBot
+    {
+        private DialogSet _dialogs;
+
+        public SimplePromptBot(BotAccessors accessors)
+        {
+            _dialogs = new DialogSet(accessors.ConversationDialogState);
+            _dialogs.Add(new TextPrompt("name"));
+        }
+
+        /// <summary>
+        /// This controls what happens when an <see cref="Activity"/> gets sent to the bot.
+        /// </summary>
+        /// <param name="turnContext">Provides the <see cref="ITurnContext"/> for the turn of the bot.</param>
+        /// <param name="cancellationToken" >(Optional) A <see cref="CancellationToken"/> that can be used by other objects
+        /// or threads to receive notice of cancellation.</param>
+        /// <returns>>A <see cref="Task"/> representing the operation result of the Turn operation.</returns>
+        public async Task OnTurnAsync(ITurnContext turnContext, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            if (turnContext == null)
+            {
+                throw new ArgumentNullException(nameof(turnContext));
+            }
+
+            // We are only interested in Message Activities.
```

Comment:
Why are we only interested in the ActivityType of message?

Reply:
here yes